File: 项目启动/NCdesktop/src/types/asset.ts
import type { Tag } from "./common";

/** 素材 — 所有类型的知识碎片 */
export interface Asset {
  id: string;
  projectId: string;
  type: AssetType;
  /** 工作区内展示名（可被 AI 重命名） */
  name: string;
  /** 拖入时的原始文件名；副本在应用目录内整理，原件路径不会被改写 */
  originalName?: string;
  filePath: string;
  fileSize: number;
  mimeType: string;
  tags: Tag[];
  capturedAt: string;
  importedAt: string;
  /** 后端 `sourceData`：如悬浮窗拖入时为原件绝对路径 */
  sourceData?: string | null;
  source: AssetSource;
  aiAnalysis: AIAnalysis | null;
  isStarred: boolean;
}

export type AssetType =
  | "photo"
  | "scan_text"
  | "audio_clip"
  | "pdf"
  | "webpage"
  | "markdown"
  | "image"
  | "other";

export type AssetSource =
  | { type: "tf_card_camera" }
  | { type: "tf_card_scanner" }
  | { type: "tf_card_mic" }
  | { type: "dropzone_drag" }
  | { type: "dropzone_paste" }
  | { type: "manual_import" };

/** AI 分析结果 */
export interface AIAnalysis {
  summary: string;
  topics: string[];
  ocrText: string | null;
  language: string;
  suggestedTags: string[];
  suggestedName: string;
}

File: 项目启动/NCdesktop/src-tauri/src/models/asset.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Asset {
    pub id: String,
    pub project_id: String,
    pub asset_type: String,
    /// 工作区内展示名（可被 AI / 用户重命名）
    pub name: String,
    /// 拖入时的原始文件名，仅副本在应用目录内被整理，此字段用于对照原件
    #[serde(default)]
    pub original_name: String,
    pub file_path: String,
    pub file_size: i64,
    pub mime_type: String,
    pub captured_at: String,
    pub imported_at: String,
    pub source_type: String,
    pub source_data: Option<String>,
    pub is_starred: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AIAnalysisRow {
    pub id: String,
    pub asset_id: String,
    pub summary: String,
    pub topics: String,
    pub ocr_text: Option<String>,
    pub language: String,
    pub suggested_tags: String,
    pub suggested_name: String,
}
